GCC warning: in 'duk__enc_value': declaration of 'h' shadows ... [-Wshadow]

```
duk_bi_json.c: In function 'duk__enc_value':
duk_bi_json.c:2115:16: warning: declaration of 'h' shadows a previous local [-Wshadow] duk_bi_json.c:1934:15: warning: shadowed declaration is here [-Wshadow]
duk_bi_json.c:2122:16: warning: declaration of 'h' shadows a previous local [-Wshadow] duk_bi_json.c:1934:15: warning: shadowed declaration is here [-Wshadow]
```

This is with version 1.5.0 ( 83d557704ee63f68ab40b6fcb00995c9b3d6777c ). Not sure when it started but IIRC I was not seeing this with earlier versions.

GCC is 4.9.3, cross compiler on host Linux 64, with target Windows 32 (MXE).

I tried looking at `duk_bi_json.c`, but the sources I use are only `duk_config.h`, `duktape.c`, `duktape.h`, `metadata.json`.

--- a/Makefile
+++ b/Makefile
@@ -272,6 +272,7 @@ CCOPTS_SHARED += -pedantic -ansi -std=c99 -fstrict-aliasing
 # -Wextra is very picky but catches e.g. signed/unsigned comparisons
 CCOPTS_SHARED += -Wall -Wextra -Wunused-result -Wdeclaration-after-statement
 CCOPTS_SHARED += -Wcast-qual
+CCOPTS_SHARED += -Wshadow
 CCOPTS_SHARED += -Wunreachable-code  # on some compilers unreachable code is an error
 # -Wfloat-equal is too picky, there's no apparent way to compare floats
 # (even when you know it's safe) without triggering warnings

--- a/RELEASES.rst
+++ b/RELEASES.rst
@@ -1649,6 +1649,9 @@ Planned
 
 * Add an extra module providing a minimal 'console' binding (GH-767)
 
+* Fix a harmless compilation warning related to a shadowed variable (GH-793,
+  GH-794)
+
 * Internal change: rework shared internal string handling so that shared
   strings are plain string constants used in macro values, rather than
   being declared as actual symbols; this reduces compilation warnings with

--- a/src/duk_bi_json.c
+++ b/src/duk_bi_json.c
@@ -1931,7 +1931,7 @@ DUK_LOCAL void duk__enc_array(duk_json_enc_ctx *js_ctx) {
 DUK_LOCAL duk_bool_t duk__enc_value(duk_json_enc_ctx *js_ctx, duk_idx_t idx_holder) {
 	duk_context *ctx = (duk_context *) js_ctx->thr;
 	duk_hthread *thr = (duk_hthread *) ctx;
-	duk_hobject *h;
+	duk_hobject *h_tmp;
 	duk_tval *tv;
 	duk_tval *tv_holder;
 	duk_tval *tv_key;
@@ -1953,12 +1953,12 @@ DUK_LOCAL duk_bool_t duk__enc_value(duk_json_enc_ctx *js_ctx, duk_idx_t idx_hold
 
 	DUK_DDD(DUK_DDDPRINT("value=%!T", (duk_tval *) duk_get_tval(ctx, -1)));
 
-	h = duk_get_hobject_or_lfunc_coerce(ctx, -1);
-	if (h != NULL) {
+	h_tmp = duk_get_hobject_or_lfunc_coerce(ctx, -1);
+	if (h_tmp != NULL) {
 		duk_get_prop_stridx(ctx, -1, DUK_STRIDX_TO_JSON);
-		h = duk_get_hobject_or_lfunc_coerce(ctx, -1);  /* toJSON() can also be a lightfunc */
+		h_tmp = duk_get_hobject_or_lfunc_coerce(ctx, -1);  /* toJSON() can also be a lightfunc */
 
-		if (h != NULL && DUK_HOBJECT_IS_CALLABLE(h)) {
+		if (h_tmp != NULL && DUK_HOBJECT_IS_CALLABLE(h_tmp)) {
 			DUK_DDD(DUK_DDDPRINT("value is object, has callable toJSON() -> call it"));
 			/* XXX: duk_dup_unvalidated(ctx, -2) etc. */
 			duk_dup(ctx, -2);         /* -> [ ... key val toJSON val ] */
@@ -1991,6 +1991,8 @@ DUK_LOCAL duk_bool_t duk__enc_value(duk_json_enc_ctx *js_ctx, duk_idx_t idx_hold
 
 	tv = DUK_GET_TVAL_NEGIDX(ctx, -1);
 	if (DUK_TVAL_IS_OBJECT(tv)) {
+		duk_hobject *h;
+
 		h = DUK_TVAL_GET_OBJECT(tv);
 		DUK_ASSERT(h != NULL);
 

